tracing/src/main/java/com/palantir/tracing/DeferredTracer.java

```diff
@@ -49,26 +65,48 @@ public DeferredTracer(String operation) {
         this(Optional.of(operation));
     }
 
-    DeferredTracer(Optional<String> operation) {
-        this.trace = Tracer.copyTrace();
-        this.operation = operation;
+    /**
+     * Create a new deferred tracer, optionally specifying an operation.
+     * If no operation is specified, will attempt to use the parent span's operation name.
+     */
+    public DeferredTracer(Optional<String> operation) {
+        Optional<Trace> maybeTrace = Tracer.copyTrace();
+        if (maybeTrace.isPresent()) {
+            Trace trace = maybeTrace.get();
+            this.traceId = trace.getTraceId();
+            this.isObservable = trace.isObservable();
+            this.parentSpanId = trace.top().map(OpenSpan::getSpanId).orElse(null);
+            this.operation = operation.orElse(trace.top().map(OpenSpan::getOperation).orElse(UNKNOWN_OPERATION));
+        } else {
+            this.traceId = null;
+            this.isObservable = false;
+            this.parentSpanId = null;
+            this.operation = null;
+        }
     }
 
     /**
      * Runs the given callable with the current trace at
      * the time of construction of this {@link DeferredTracer}.
      */
     public <T, E extends Throwable> T withTrace(Tracers.ThrowingCallable<T, E> inner) throws E {
-        if (!trace.isPresent()) {
+        if (traceId == null) {
```

Comment:
if we specify an operation name, but haven't started a trace yet, the deferred tracer will not create a new trace or span. Is that expected?

Reply:
I suppose it matches the previous logic, so not something we should focus on in this PR.

Reply:
yeah, not sure - it matches the previous behaviour